enhancements/multi-arch/multiarch-manager-operator.md

```diff
@@ -0,0 +1,888 @@
+
+#### Required RBAC permissions
+
+The service account running the operator components will need the following RBAC
+permissions:
+
+- Deployment (namespace-scoped)
+  - Watch, List, Get, Create, Update, Delete
+- Pod (cluster-scoped)
+  - Watch, List, Get, Update
+- Secrets (cluster-scoped)
+  - Watch, List, Get
+- ConfigMaps (MMO operator namespace and openshift-image-registry)
+  - Watch, List, Get
+- PodPlacementConfig CRD
+  - Watch, List, Get, Update
+- admissionregistration.k8s.io/MutatingWebhookConfiguration
+  - Get, Update
+- image.openshift.io/ImagesStream/layers (required for internal registry access)
+  - Get
+- operator.openshift.io/ImageContentSourcePolicy
+- config.openshift.io/ImageDigestMirrorSet
+- config.openshift.io/ImageTagMirrorSet
+  - Watch, List, Get
+- config.openshift.io/Image
+  - Watch, List, Get
+
+#### PodPlacementConfig CR
+The configuration CR of the operator allows the admins to set the following:
+- a `NamespaceSelector` to filter the namespaces MMO should patch the pods of.
+The `openshift-*`, `kube-*` and `hypershift-*` namespaces are implicitly excluded.
+- the `LogVerbosity` level of the operator
+A new Custom Resource Definition (CRD) is being proposed to handle the
+configuration of the operator deployment.
+
+```go
+// +kubebuilder:validation:Enum=Normal;Debug;Trace;TraceAll
+type LogVerbosityLevel string
+
+const (
+    LogVerbosityLevelNormal LogVerbosityLevel = "Normal"
+    LogVerbosityLevelDebug LogVerbosityLevel = "Debug"
+    LogVerbosityLevelTrace LogVerbosityLevel = "Trace"
+    LogVerbosityLevelTraceAll LogVerbosityLevel = "TraceAll"
+)
+
+
+// PodPlacementConfigSpec defines the desired state of PodPlacementConfig
+type PodPlacementConfigSpec struct {
+    // logVerbosity is the log level for the pod placement controller
+    // Valid values are: "Normal", "Debug", "Trace", "TraceAll".
+    // Defaults to "Normal".
+    // +optional
+    // +kubebuilder:default=Normal
+    LogVerbosity LogVerbosityLevel `json:"logVerbosity,omitempty"`
+
+    // namespaceSelector is a metav1.LabelSelector to filter the namespaces whose
+    // pods should be patched with the scheduling gate at admission time.
+    // Any pod in a namespace not matched by the selector will not be controlled by the operator.
+    // The webhook will always ignore pods in reserved namespaces matching the
+    // openshift-*, kube-*, and hypershift-* globs.
+    NamespaceSelector metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+
+}
+
+// PodPlacementConfigStatus defines the observed state of PodPlacementConfig
+type PodPlacementConfigStatus struct {
+    // Conditions represents the latest available observations of a PodPlacementConfig's current state.
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// PodPlacementConfig is the Schema for the podplacementconfigs API
+type PodPlacementConfig struct {
```

Comment:
A few related questions:
- I assume this is expected to be a singleton CR since it controls the verbosity level of the controller? If so, I'd suggest including CEL-based validation for a specific `metadata.name`.
- Is this CRD cluster-scoped or namespace-scoped? I assume cluster-scoped since its configuration results in behavior changes across other namespaces?

Reply:
Example of CEL-based singleton validation for a cluster-scoped object here: https://github.com/openshift/api/blob/a7ca92db022ae3ccf29cfa8865e3e35e71013de2/operator/v1alpha1/types_olm.go#L17

Reply:
Yes - the CR is a singleton and cluster scoped. I have updated the doc to reflect that and also added the CEL-based validation.